include/alpaka/intrinsic.hpp

```diff
@@ -9,13 +9,28 @@
 #include "alpaka/api/trait.hpp"
 #include "internal/intrinsic.hpp"
 
+#include <climits>
+
 namespace alpaka
 {
-    constexpr auto popcount(auto const& arg)
+    /** Returns the number of bits set to 1. */
+    constexpr int32_t popcount(auto const& arg)
+        requires(sizeof(ALPAKA_TYPEOF(arg)) == 4u || sizeof(ALPAKA_TYPEOF(arg)) == 8u)
     {
-        auto const intrinsicImpl = alpaka::trait::getIntrinsicImpl(thisApi());
+        constexpr auto const intrinsicImpl = alpaka::trait::getIntrinsicImpl(thisApi());
```

Comment:
```suggestion
        constexpr auto intrinsicImpl = alpaka::trait::getIntrinsicImpl(thisApi());
```
It is not a problem but just pointing out that constexpr implies const. This also happens below.

Reply:
fixed